src/tcp_address.cpp

```diff
@@ -622,33 +647,6 @@ int zmq::tcp_address_t::resolve (const char *name_, bool local_, bool ipv6_, boo
           addr_str [addr_str.size () - 1] == ']')
         addr_str = addr_str.substr (1, addr_str.size () - 2);
 
-    // Test the '%' to know if we have an interface name / zone_id in the address
```

Comment:
zone_id is expected in the passed _name string for non-global/link-local IPv6 addresses. Without the zone_id, a link-local IPv6 address can not be resolved with no ambiguity. Linux OS expects the sin6_scope_id to be set appropriately for a link-local address for it to bind a socket successfully. sin6_scope_id is not always '0' and it depends on the interface and its index in the system. This code was necessary to get zeromq sockets work properly on IPv6 link local addresses.